src/object/srv_obj.c

```diff
@@ -41,26 +41,23 @@
 #include "obj_rpc.h"
 #include "obj_internal.h"
 
-static inline struct obj_tls *
-obj_tls_get()
-{
-	return dss_module_key_get(dss_tls_get(), &obj_module_key);
-}
-
 /**
  * After bulk finish, let's send reply, then release the resource.
  */
 static void
 ds_obj_rw_complete(crt_rpc_t *rpc, struct ds_cont_hdl *cont_hdl,
 		   daos_handle_t ioh, int status, uint32_t map_version)
 {
+	struct obj_tls		*tls = obj_tls_get();
```

Comment:
srv_profile is just the profile chunk list header. since we need allocation for profile anyway, so I would rather put a pointer in TLS to avoid extra allocation for object tls if there are no profile needed. Not sure what you mean profile_start to switch one bit?